documents/trail/ecs-mini-program.yml

```diff
@@ -0,0 +1,353 @@
+ROSTemplateFormatVersion: '2015-09-01'
+Parameters:
+  InstanceId:
+    Type: String
+    Label:
+      en: ECS Instance ID
+      zh-cn: ECS实例ID
+    AssociationProperty: 'ALIYUN::ECS::Instance::InstanceId'
+  EcsRamRoleName:
+    Type: String
+    Label:
+      en: RAM Role Name
+      zh-cn: RAM角色名称
+Resources:
+  DS_Instances:
+    Type: 'DATASOURCE::ECS::Instances'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+  WaitConditionHandle:
+    Type: 'ALIYUN::ROS::WaitConditionHandle'
+    Properties: {}
+  WaitCondition:
+    Type: 'ALIYUN::ROS::WaitCondition'
+    Properties:
+      Handle:
+        Ref: WaitConditionHandle
+      Timeout: 3720
+      Count: 1
+  EcsRamRole:
+    Type: 'ALIYUN::RAM::Role'
+    Properties:
+      RoleName:
+        Ref: EcsRamRoleName
+      Description: 搭建支付宝小程序ECS实例角色
+      AssumeRolePolicyDocument:
+        Version: '1'
+        Statement:
+          - Action: 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              Service:
+                - ecs.aliyuncs.com
+      Policies:
+        - PolicyName:
+            'Fn::Sub':
+              - '${RamRoleName}Policy'
+              - RamRoleName:
+                  Ref: EcsRamRoleName
+          Description: 搭建支付宝小程序ECS实例角色权限策略
+          PolicyDocument:
+            Version: '1'
+            Statement:
+              - Action:
+                  - 'ecs:DescribeInstances'
+                  - 'ecs:DescribeInstanceStatus'
+                Effect: Allow
+                Resource:
+                  - '*'
+  EcsRamRoleAttachment:
+    Type: 'ALIYUN::ECS::RamRoleAttachment'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      RamRoleName:
+        Ref: EcsRamRoleName
+    DependsOn: EcsRamRole
+  SecurityGroupIngress_80:
+    Type: 'ALIYUN::ECS::SecurityGroupIngress'
+    Properties:
+      SecurityGroupId:
+        'Fn::Jq':
+          - First
+          - '.[0].SecurityGroupIds[0]'
+          - 'Fn::GetAtt':
+              - DS_Instances
+              - Instances
+      SourceCidrIp: 0.0.0.0/0
+      IpProtocol: tcp
+      NicType: intranet
+      PortRange: 80/80
+  InstallWebServer:
+    Type: 'ALIYUN::ECS::RunCommand'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      Type: RunShellScript
+      Sync: true
+      Timeout: 3600
+      CommandContent:
+        'Fn::Sub': >-
+          #!/bin/bash
+
+
+          if [ ! -f .ros.provision ]; then
+            echo "Name: 搭建支付宝小程序" > .ros.provision
+          fi
+
+
+
+          name=$(grep "^Name:" .ros.provision | awk -F':' '{print $2}' | tr -d
```

Comment:
Please use `name=$(grep "^Name:" .ros.provision | awk -F':' '{print $2}' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')`. Because this method only strips the spaces on both sides.

You can refer to [this](https://github.com/aliyun/ros-templates/blob/master/documents/trail/ecs-lnmp.yml#L100).